runtime/lua/vim/fs.lua

```diff
@@ -143,6 +146,13 @@ function M.dir(path, opts)
   end)
 end
 
+--- @class vim.fs.find.opts
+--- @field path string
+--- @field upward boolean
+--- @field stop string
+--- @field type string
+--- @field limit number
```

Comment:
Shouldn't some of these be nil?

Reply:
Ah, good point.

I'm not sure why I'm not getting diagnostics for these.

Reply:
Ah, it's because we are casting into this type so they aren't needed. If the type was used at `@param` then we'd need to makes these optional.